deployment/katanaDeployment.js

```diff
@@ -0,0 +1,135 @@
+      if (byteArray.length > 0 && byteArray[0] === 3) {
+        console.log('^C');
+        process.exit(1);
+      }
+      process.stdin.setRawMode(false);
+      resolve();
+    })
+  );
+};
+
+async function pressToContinue() {
+    console.log("Checkpoint... Press any key to continue!");
+    await keypress();
+  }
+
+let kncAddress;
+let deployerAddress;
+let epochPeriod;
+let starttime;
+let shortExecutorConfig;
+let longExecutorConfig;
+let daoOperator;
+let outputFilename;
+
+task('deployGovInfra', 'deploys staking, governance, voting power strategy and executors').setAction(async () => {
+  parseInput(configParams);
+  const [deployer] = await ethers.getSigners();
+  deployerAddress = await deployer.getAddress();
+
+  // contract deployment
+  console.log('deploying staking contract...');
+  const KyberStaking = await ethers.getContractFactory('KyberStaking');
+  const kyberStaking = await KyberStaking.deploy(deployerAddress, kncAddress, epochPeriod, starttime);
+  await kyberStaking.deployed();
+  console.log(`staking address: ${kyberStaking.address}`);
+  await pressToContinue();
+
+  console.log('deploying governance contract...');
+  const KyberGovernance = await ethers.getContractFactory('KyberGovernance');
+  const kyberGovernance = await KyberGovernance.deploy(deployerAddress, daoOperator, [], []);
+  await kyberGovernance.deployed();
+  console.log(`governance address: ${kyberGovernance.address}`);
+  await pressToContinue();
+
+  console.log('deploying short executor...');
+  const Executor = await ethers.getContractFactory('DefaultExecutor');
+  const shortExecutor = await Executor.deploy(
+    kyberGovernance.address,
+    shortExecutorConfig.delay,
+    shortExecutorConfig.gracePeriod,
+    shortExecutorConfig.minimumDelay,
+    shortExecutorConfig.maximumDelay,
+    shortExecutorConfig.minVoteDuration,
+    shortExecutorConfig.maxVotingOptions,
+    shortExecutorConfig.voteDifferential,
+    shortExecutorConfig.minimumQuorum
+  );
+  await shortExecutor.deployed();
+  console.log(`shortExecutor address: ${shortExecutor.address}`);
+  await pressToContinue();
+
+  console.log('deploying long executor...');
+  const longExecutor = await Executor.deploy(
+    kyberGovernance.address,
+    longExecutorConfig.delay,
+    longExecutorConfig.gracePeriod,
+    longExecutorConfig.minimumDelay,
+    longExecutorConfig.maximumDelay,
+    longExecutorConfig.minVoteDuration,
+    longExecutorConfig.maxVotingOptions,
+    longExecutorConfig.voteDifferential,
+    longExecutorConfig.minimumQuorum
+  );
+  await longExecutor.deployed();
+  console.log(`longExecutor address: ${longExecutor.address}`);
+  await pressToContinue();
+
+  console.log('deploying voting power strat...');
+  const VotingPowerStrategy = await ethers.getContractFactory('EpochVotingPowerStrategy');
+  const votingPowerStrategy = await VotingPowerStrategy.deploy(
+    kyberGovernance.address,
+    kyberStaking.address
+  );
+  await votingPowerStrategy.deployed();
```

Comment:
Well actually, you can await multiple contracts are deployed at the same time, by using `Promise.all`

Reply:
that is true, but from mainnet experience, better to do and monitor 1 by 1, gas price fluctuates very quickly.. maybe should ping user for input for gas price to be used for the next tx

Reply:
updated to be as such